source/renderer/app/analytics/MatomoClient.ts

```diff
@@ -0,0 +1,65 @@
+import MatomoTracker from 'matomo-tracker';
+import { AnalyticsClient } from './types';
+import { Environment } from '../../../common/types/environment.types';
+import formatCpuInfo from '../utils/formatCpuInfo';
+import {
+  ANALYTICS_API_ENDPOINT,
+  DEV_MODE_SITE_MAP_ID,
+  NETWORK_TO_ANALYTICS_SITE_ID_MAP,
+} from '../config/analyticsConfig';
+
+const CPU_DIMENSION_KEY = 'dimension2';
+const RAM_DIMENSION_KEY = 'dimension3';
+const OS_DIMENSION_KEY = 'dimension4';
+const VERSION_DIMENSION_KEY = 'dimension5';
+
+/**
+ * Matomo API reference:
+ * https://developer.matomo.org/api-reference/tracking-api
+ */
+export class MatomoClient implements AnalyticsClient {
+  private matomoTracker: MatomoTracker;
+
+  constructor(private environment: Environment, private userId: string) {
+    this.matomoTracker = new MatomoTracker(
+      this.getMatomoSiteId(environment),
+      ANALYTICS_API_ENDPOINT
+    );
+  }
+
+  sendPageNavigationEvent = async (pageTitle: string) => {
+    this.matomoTracker.track({
+      _id: this.userId,
+      action_name: pageTitle,
+      url: this.getAnalyticsURL(),
+      [CPU_DIMENSION_KEY]: formatCpuInfo(this.environment.cpu),
+      [RAM_DIMENSION_KEY]: Math.round(
```

Comment:
I suggest extracting this and keeping it in utils.

Reply:
Good suggestion, and actually we already have a utility to convert from bytes to human friendly value so I reused it.